packages/lab/src/radio-button/RadioButton.tsx

```diff
@@ -19,34 +17,21 @@ export interface RadioButtonProps
   checked?: boolean;
   className?: string;
   disabled?: boolean;
-  /**
-   * Custom icon component
-   */
-  icon?: ComponentType<RadioButtonIconProps>;
   /**
    * The label to be shown next to the radio
    */
-  label?: ControlLabelProps["label"];
-  LabelProps?: Partial<ControlLabelProps>;
+  label?: ReactNode;
   name?: string;
   onChange?: ChangeEventHandler<HTMLInputElement>;
   value?: string;
 }
 
 export const RadioButton = forwardRef<HTMLLabelElement, RadioButtonProps>(
   function RadioButton(props, ref) {
-    const {
-      checked,
-      className,
-      icon,
-      disabled,
-      label,
-      LabelProps,
-      value,
-      onChange,
-      ...rest
-    } = props;
+    const { checked, className, disabled, label, value, onChange, ...rest } =
+      props;
```

Comment:
I had a conversation with David and Immanuel last week about something important with these controls. 

Is the current pattern of refs and props being spread on the root the right one? With these controls the most important element is the input

Reply:
That makes sense actually. I'll make the changes. Thanks

Reply:
I think we should speak about it first :)